### PR TITLE
Improved Request Throttling and Added the option to switch between Fuzzy search and Direct search

### PR DESCRIPTION
**Pull request by penwoodjon**
## Steps to Test
1. Review code changes for request throttling and verify they make sense in this context.
2. Install and test the integration with: `45.139.236.58 and polarity.io` and verify the responses make sense.
3. Search `45.139.236.58` with and without the Direct Search Option selected and verify with Direct Search not selected you get results and with it selected you get no results.

### CODE REVIEW

File: integration.js [hunk 1]
```diff
@@ -1,314 +1,341 @@
...
+  let hostUrl = url.parse(options.host);
+
+  request(
+    {
+      url: `${options.authHost}/${options.tenantId}/tokens/OAuth/2`,
+      formData: {
+        grant_type: "client_credentials",
+        client_id: `${options.clientId}@${options.tenantId}`,
+        client_secret: options.clientSecret,
+        resource: `00000003-0000-0ff1-ce00-000000000000/${hostUrl.host}@${options.tenantId}`
```
Comment (sarus) [inline integration.js]: I see this didn't change but is that resource id `00000003-0000-0ff1-ce00-000000000000` supposed to be hard coded?

Comment (penwoodjon) [reply integration.js]: I'm not sure. @radar989 ?

Comment (sarus) [reply integration.js]: Actually just looked it up.  It's the audience principal ID and is fixed.

**Comment by penwoodjon** [comment]
> @penwoodjon Need to increment the package.json file to 3.1.0 to update the version number.

:thumbsup:

**Pull request merged**